pkg/twcc/twcc.go

```diff
@@ -19,7 +20,7 @@ type pktInfo struct {
 // transport wide congestion control feedback reports as specified in
 // https://datatracker.ietf.org/doc/html/draft-holmer-rmcat-transport-wide-cc-extensions-01
 type Recorder struct {
-	receivedPackets []pktInfo
+	receivedPackets map[uint32]pktInfo
```

Comment:
Based on very limited `golang` knowledge, `map`s seem to incur more memory (expected, but making the comment in the sense I have been surprised some times maps growing unexpectedly). Is there a need to use map here? 

Also, this is constructing the slice from map in building feedback packet. That can be avoided I think if we just append and sort that slice directly.

Reply:
@boks1971 thanks for the comment! I used a map here to handle the need to de-dupe packets with the same sequence number (see the previous `insertSorted` method). I could use a slice and de-dupe during iteration when building the feedback packet, but this would require a map/set anyway

Reply:
Thank you @treyhakanson . As these maps should be small, it should be fine. On that one comment, I think re-allocating the `map` after building report would be needed to ensure that map does not grow. Did not see it. Maybe, I missed it in the diff.

Reply:
You can do the deduplication when building the feedback packet without a map. When iterating over the sorted slice, you remember the previous value and compare it to the next because in the sorted slice, all duplicates will be neighbors. It could be somewhat fiddly, depending on whether you want to insert the first or last occurrence. Currently, we take the last (which would be more complicated), but I think we may want to use the first instead (ordered by arrival time, though, not by the time it was added). That could be added as a criterion to the sort function, then it would be easy again (just check if the current sequence number was already added in the last iteration). However, all this is probably very hypothetical since there will rarely be duplicates anyway.

Reply:
Ah you're totally right, not sure why I didn't think of that; thanks for calling out!
Updated to implement this approach and take the packet with the earliest arrival time. I also updated `TestDuplicatePackets` to test that we take the packet with the earliest arrival time even if it isn't the first one we record. Fwiw, I've been testing on LTE/5G networks recently and have noticed a fair bit of duplication so its seems worth handling

Reply:
Oh, that's good to know, thanks!
And thanks for implementing this change!